docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,446 @@
+
+A few often-used architectures are listed below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Architecture
+     - AMD GPU name
+   * - gfx90a
+     - AMD Instinct(TM) MI210/250/250X Accelerator
+   * - gfx908
+     - AMD Instinct(TM) MI 100 Accelerator
+   * - gfx906
+     - | AMD Radeon Instinct(TM) MI50/60 Accelerator
+       | AMD Radeon(TM) (Pro) VII Graphics Card
+   * - gfx900
+     - | Radeon Instinct(TM) MI 25 Accelerator
+       | Radeon(TM) RX Vega 64/56 Graphics
+
+For a host with ROCm installed, the device architecture can be retrieved via the ``rocminfo`` tool.
+The architecture will be displayed in the ``Name:`` row.
+
+.. _build_for_portlibs_dpcpp:
+
+Pure SYCL backends: portBLAS and portFFT
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`portBLAS <https://github.com/codeplaysoftware/portBLAS>`_ and 
+`portFFT <https://github.com/codeplaysoftware/portFFT>`_ are 
+experimental pure-SYCL backends that work on all
+SYCL targets supported by the DPC++ compiler. Since they support multiple targets,
+they cannot be enabled with other backends in the same domain.
+Both libraries are experimental and currently only support a subset of operations
+and features.
+
+For best performance, both libraries must be tuned. See the individual sections 
+for more details.
+
+Both portBLAS and portFFT are used as header-only libraries, and will be downloaded
+automatically if not found.
+
+.. _build_for_portblas_dpcpp:
+
+Building for portBLAS
+---------------------
+
+`portBLAS <https://github.com/codeplaysoftware/portBLAS>`_ is
+enabled by setting ``-DENABLE_PORTBLAS_BACKEND=ON``.
+
+By default, the portBLAS backend is not tuned for any specific device.
+This tuning is required to achieve best performance.
+portBLAS can be tuned for a specific hardware target by adding compiler
+definitions in 2 ways:
+
+#.
+  Manually specify a tuning target with ``-DPORTBLAS_TUNING_TARGET=<target>``.
+  The list of portBLAS targets can be found
+  `here <https://github.com/codeplaysoftware/portBLAS#cmake-options>`_.
+  This will automatically set ``-fsycl-targets`` if needed.
+#.
+  If one target is set via ``-fsycl-targets`` the configuration step will
+  try to automatically detect the portBLAS tuning target. One can manually
+  specify ``-fsycl-targets`` via ``CMAKE_CXX_FLAGS``. See
+  `DPC++ User Manual <https://intel.github.io/llvm-docs/UsersManual.html>`_
+  for more information on ``-fsycl-targets``.
+
+
+.. _build_for_portfft_dpcpp:
+
+Building for portFFT
+---------------------
+
+`portFFT <https://github.com/codeplaysoftware/portFFT>`_ is
+enabled by setting ``-DENABLE_PORTFFT_BACKEND=ON``.
+
+By default, the portFFT backend is not tuned for any specific device. The tuning flags are
+detailed in the `portFFT <https://github.com/codeplaysoftware/portFFT>`_ repository, and can 
+set at configuration time.
+Note that some tuning configurations may be incompatible
+with some targets.
```

Comment:
Nitpicking, some sentences are split in multiple lines for no apparent reasons in this section.

Reply:
I've formatted for 80 character lines in [7b69703](https://github.com/oneapi-src/oneMKL/pull/510/commits/7b69703e3f90eedd24e90b805ff18ac78a7d6778) and [73d351](https://github.com/oneapi-src/oneMKL/pull/510/commits/73d3510b39a2b5973378e67486fa9634c8ff6697).